src/styles/components/_cell.scss

```diff
@@ -0,0 +1,60 @@
+.cell {
+  box-sizing: border-box;
+  width: $day-size;
+  height: $day-size;
+  position: relative;
+  background-color: $color-day-main;
+  border: $day-border;
+  transition: all 0.5s;
+
+  &:hover {
+    background-color: $color-day-hover;
+    transform: translateY(-20px);
+  }
+
+  @for $i from 1 through $num-days {
+    &--#{$i}::before {
+      content: "#{$i}";
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      font-family: $day-font;
+      font-size: $day-font-size;
+      color: $color-day-number;
+    }
+  }
+
+  &--1 {
+    @if $start-week-day == mon {
+      margin-left: 0;
+    }
+    @else if $start-week-day == teu {
+      margin-left: ($day-size + $cell-gap) * 1;
+    }
+    @else if $start-week-day == wed {
+      margin-left: ($day-size + $cell-gap) * 2;
+    }
+    @else if $start-week-day == thu {
+      margin-left: ($day-size + $cell-gap) * 3;
+    }
+    @else if $start-week-day == fri {
+      margin-left: ($day-size + $cell-gap) * 4;
+    }
+    @else if $start-week-day == sat {
+      margin-left: ($day-size + $cell-gap) * 5;
+    }
+    @else if $start-week-day == sun {
+      margin-left: ($day-size + $cell-gap) * 6;
+    }
+    @else {
+      margin-left: 0;
+    }
+  }
```

Comment:
Don't use position modifiers directly to first cell. Use parent modifier with for instance ` :nth-child(-n + 1)` and it's better to do this uses @each to create modifiers.